File: parser.js
const fs = require('fs');

const parseData = (fileData) => {
    const content = fileData.split('\n');
    const validList = {}
    true_values = ['on', 'yes', 'true']
    false_values = ['off', 'no', 'false']

    for (let i = 0; i < content.length; i++) {
        let l = content[i];
        if (l !== '' && l[0] !== '#') {
            let pair = l.split('=');
            if (pair.length === 2) {
                let key = pair[0].trim();
                let value = pair[1].trim();
                if(!isNaN(value)){
                    validList[key] = parseFloat(value);
                    continue;
                }
                if (true_values.includes(value)) {
                    validList[key] = true
                    continue;
                }
                if (false_values.includes(value)) {
                    validList[key] = false;
                    continue;
                }
                validList[key] = value
            }
        }
    }

    return JSON.stringify(validList)
}

const readFile = (file) => {
    return new Promise((resolve, reject) => {
        fs.readFile(file, 'utf8', (err, data) => {
            if (err) {
                console.error(err)
                reject(err)
            }
            resolve(data)
        })
    })
}


const parseConfig = (filename) => {
    readFile(filename).then((data)=>{
        console.log(parseData(data));
    })
} 

parseConfig('sample1.cfg')
parseConfig('sample2.cfg')

File: parser.py

def isfloat(str):
    try:
        float(str)
        res = True
    except:
        res = False
    return res

def parse_config(filename):

    with open(filename) as f:
        content = f.readlines()

    true_values = ['on', 'yes', 'true']
    false_values = ['off', 'no', 'false']

    valid_list = {}

    for line in content:
        l = line.rstrip('\n')
        if l != '' and l[0] != '#' and '=' in l:

            pair = l.split('=') 
            key = pair[0].strip(' ')
            val = pair[1].strip(' ')

            if val.isnumeric():
                valid_list[key] = int(val)
                continue
            
            if isfloat(val):
                valid_list[key] = float(val)
                continue

            if val in true_values:
                valid_list[key] = True
                continue

            if val in false_values:
                valid_list[key] = False
                continue

            valid_list[key] = val
            
    return valid_list 

if __name__ == "__main__":
    CONFIG1 = parse_config('sample1.cfg')
    print(CONFIG1)

    CONFIG2 = parse_config('sample2.cfg')
    print(CONFIG2)
